src/main/java/seedu/address/model/lab/LabList.java

```diff
@@ -187,6 +187,49 @@ public boolean isEmpty() {
         return internalList.isEmpty();
     }
 
+    /**
+     * Aligns the current LabList to the given {@code toAlignWith}.
+     * Aligning LabList means that the current LabList has to have all the Labs that the {@code toAlignWith} has,
+     * and no additional Labs that the {@code toAlignWith} doesn't have.
+     * The end result is that both LabList will have the same list of Labs with potentially different status and marks.
+     * Most relevant for deserialization of labs in {@code JsonSerializableAddressBook}.
+     *
+     * @param toAlignWith The LabList that you want to align to.
+     * @throws DuplicateLabException if one of the LabList contains duplicate labs. (should not be the case)
+     */
+    public void alignLabs(LabList toAlignWith) throws DuplicateLabException {
+        requireNonNull(toAlignWith);
+        // Using LabList instead of List<Lab> for O(n) setLabs instead of O(n^2) setLabs
+        LabList replacementList = new LabList();
+
+        // Iterator variable for toAlignWith
+        int i = 0;
```

Comment:
Perhaps it would be better to use more informative names for `i` and `j` to make it easier to reason about the behavior of the while loop?

Reply:
Good suggestion!